app/script/conversation/ConversationEphemeralHandler.js

```diff
@@ -30,48 +30,65 @@ z.conversation.ConversationEphemeralHandler = class ConversationEphemeralHandler
   }
 
   constructor(conversationService, onMessageTimeout) {
+    this._updateTimedMessages = this._updateTimedMessages.bind(this);
+
     this.conversationService = conversationService;
     this.onMessageTimeout = onMessageTimeout;
     this.logger = new z.util.Logger('z.conversation.ConversationEphemeralHandler', z.config.LOGGER.OPTIONS);
 
-    this.timedMessages = [];
-    window.setInterval(this._updateTimedMessages.bind(this), ConversationEphemeralHandler.CONFIG.INTERVAL_TIME);
+    this.timedMessages = ko.observableArray([]);
+
+    let updateIntervalId = undefined;
+    this.timedMessages.subscribe(messageEntities => {
+      const shouldClearInterval = !messageEntities.length && updateIntervalId;
+      if (shouldClearInterval) {
+        return window.clearInterval(updateIntervalId);
```

Comment:
`updateIntervalId` should be set to a falsy value as well.

Reply:
Upps.. thanks for the catch